### PR TITLE
Assignment2

### PR DESCRIPTION
**Pull request by sudarson-25**

### CODE REVIEW

**Comment by Dakshita-Minocha** [comment]
@sudarson-25 Undo this merge and re-commit changes into branch A2. Assign pull request back to me.

File: Training/Training/intlist.c [hunk 1]
```diff
@@ -0,0 +1,290 @@
+#include "intlist.h"
+#include <malloc.h>
+#include<stdbool.h>
+
+// bool variable to flag deleted lists
+bool isDeleted = false;
+
+struct Node {
+	int data;
+	struct Node* next; //we can create pointers to incomplete types
```
Comment (sanjivtmm) [inline Training/Training/intlist.c:10]: What is the purpose of this comment?

File: Training/Training/intlist.c [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+/// The flag ensures whether the list exists.
+/// The function then checks if the head pointer points to a list(empty or with nodes).
+/// If it points to a node, the node's address is temporarily stored in a pointer. The head pointer is modified to point to 
+/// the next node and the previous node is freed.
+/// </summary>
+/// <param name="head"></param>
+/// <returns>Error code (if any)</returns>
+int Delete (struct Node** head) {
+
+	if (isDeleted == true) {
```
Comment (sanjivtmm) [inline Training/Training/intlist.c:51]: What if there are 2 linked lists and you deleted one? You have a single "isDeleted" flag for multiple lists. How will this work?

File: Training/Training/intlist.c [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+
+
+/// <summary>
+/// The flag ensures whether the list exists.
+/// Adds Nodes to an existing list. 
+/// </summary>
+/// <param name="head"></param>
+/// <param name="data"></param>
+/// <returns>Error code (if any)</returns>
+int Add (struct Node** head, int data) {
```
Comment (sanjivtmm) [inline Training/Training/intlist.c:79]: Why a double indirection? i.e. **?

File: Training/Training/intlist.c [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+	}
+
+	// Search for the key to be deleted, keep track of the
+	// previous node as we need to change 'prev->next'
+	while (temp != NULL && temp->data != ele) {
+		prev = temp;
+		temp = temp->next;
+	}
+
+	// If key was not present in linked list
```
Comment (sanjivtmm) [inline Training/Training/intlist.c:228]: The comment is not correct. It is conveying the opposite.

File: Training/Training/intlist.h [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+#define LIST_DOE -4
+
+struct Node* Create ();
+struct Node* CreateNode (int data);
+int Delete (struct Node** head);
+int Add (struct Node** head, int data);
+int Insert (struct Node** head, int ind, int data);
+int RemoveAt (struct Node** head, int ind);
+int Remove (struct Node** head, int ele);
+int Count (struct Node* head);
```
Comment (sanjivtmm) [inline Training/Training/intlist.h:16]: Why does the signature for Count use a single indirection while the earlier functions use double?

File: Training/Training/testintlist.c [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+	//Adding to empty list
+	printf ("Add\n");
+	printf ("Case2\n");
+	printf ("Expected: 0\n");
+	printf ("Output: %d\n\n", Add (&head, 10));
+
+	//Adding to a deleted list
+	printf ("Case3\n");
+	Delete (&head);
+	printf ("Expected: -4\n");
```
Comment (sanjivtmm) [inline Training/Training/testintlist.c:31]: What is the magic number -4?

File: Training/Training/testintlist.c [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+	//Inserting into a deleted list
+	printf ("Insert\n");
+	printf ("Case6\n");
+	printf ("Expected: -4\n");
+	printf ("Output: %d\n\n", Insert (&head, 3, 20));
+
+	//Inserting into an invalid index
+	head = Create ();
+	printf ("Case7\n");
+	printf ("Expected: -2\n");
```
Comment (sanjivtmm) [inline Training/Training/testintlist.c:59]: What is the magic number -2?

**Pull request merged**